components/script/dom/underlyingsourcecontainer.rs

```diff
@@ -126,6 +127,7 @@ impl UnderlyingSourceContainer {
                 let cx = GlobalScope::get_cx();
                 rooted!(in(*cx) let mut this_object = ptr::null_mut::<JSObject>());
                 unsafe {
+                    this_object.set(JS_NewObject(*cx, ptr::null_mut()));
```

Comment:
[`JS_NewObject`](http://devdoc.net/web/developer.mozilla.org/en-US/docs/SpiderMonkey/JSAPI_Reference/JS_NewObject.html) can fail, so we should probably add an assertion (or even fail gracefully).
```rust
let obj = JS_NewObject(*cx, ptr::null_mut());
assert!(!obj.is_null());
this_object.set(obj);
```

Reply:
Thanks for pointing it out.